examples/seismic/tutorials/13_LSRTM_acoustic.ipynb

```diff
@@ -0,0 +1,733 @@
+    "        mask.data[:,iz] = (1/(start+1-nztaper))*iz - nztaper*(1/(start+1-nztaper))\n",
+    "mask.data[:,:] = 1. - mask.data[:,:] \n",
+    "plot_image_mask(mask.data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we are going to perform the LSRTM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LSRTM Iteration 1\n",
+      "LSRTM Iteration 2\n",
+      "LSRTM Iteration 3\n",
+      "LSRTM Iteration 4\n",
+      "LSRTM Iteration 5\n",
+      "LSRTM Iteration 6\n",
+      "LSRTM Iteration 7\n",
+      "LSRTM Iteration 8\n",
+      "LSRTM Iteration 9\n",
+      "LSRTM Iteration 10\n",
+      "LSRTM Iteration 11\n",
+      "LSRTM Iteration 12\n",
+      "LSRTM Iteration 13\n",
+      "LSRTM Iteration 14\n",
+      "LSRTM Iteration 15\n",
+      "LSRTM Iteration 16\n",
+      "LSRTM Iteration 17\n",
+      "LSRTM Iteration 18\n",
+      "LSRTM Iteration 19\n",
+      "LSRTM Iteration 20\n"
+     ]
+    }
+   ],
+   "source": [
+    "#NBVAL_SKIP\n",
+    "image_up_dev = Function(name='image_up_dev', grid=model.grid)\n",
+    "nrec=101\n",
+    "niter=20 # number of iterations of the LSRTM\n",
+    "history = np.zeros((niter, 1)) #objective function\n",
+    "alfa_history = np.zeros((niter, 1)) # step-length\n",
+    "history_norm = np.zeros((niter, 1)) # normalized objective function\n",
+    "residual_niter = np.zeros((niter, 1))\n",
+    "\n",
+    "grid_iter = Grid(shape=(niter,geometry.nt, model.shape[1]))\n",
+    "\n",
+    "residual_k = Function(name='residual_k', grid=grid_iter)\n",
+    "\n",
+    "for k in range(niter) :\n",
+    "    \n",
+    "    grad_full = Function(name='grad_full', grid=model.grid)\n",
+    "    \n",
+    "    residual = Receiver(name='residual', grid=model.grid, time_range=geometry.time_axis,\n",
+    "                   coordinates=geometry.rec_positions)\n",
+    "    \n",
+    "    residual_num = Receiver(name='residual_num', grid=model.grid, time_range=geometry.time_axis,\n",
+    "                   coordinates=geometry.rec_positions)\n",
+    "          \n",
+    "    d_obs = Receiver(name='d_obs', grid=model.grid,\n",
+    "                         time_range=geometry.time_axis,\n",
+    "                         coordinates=geometry.rec_positions)\n",
+    "\n",
+    "    d_syn = Receiver(name='d_syn', grid=model.grid,\n",
+    "                         time_range=geometry.time_axis,\n",
+    "                         coordinates=geometry.rec_positions)\n",
+    "    \n",
+    "    dm =  image_up_dev.data/((model0.vp.data)**3) \n",
+    "\n",
+    "    dm_true =  (solver.model.vp.data**(-2) - model0.vp.data**(-2)) # Using devito's reference of virtual source\n",
+    "\n",
+    "    print('LSRTM Iteration',k+1)\n",
+    "    objective = 0.\n",
+    "    for i in range(nshots):\n",
```

Comment:
Probably you could improve cell 11, where the LSRTM is performed. Maybe you could replace the interior loop shots of LSRTM by a function call. The `03_FWI` notebook use the same approach.  

As you declare the vast majority of the variables within that loop, I believe that is a simple change.

Reply:
Thank you @speglich . All the changes have been made. Let me know if it works for you.